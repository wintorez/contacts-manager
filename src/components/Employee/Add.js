import React from 'react';
import PropTypes from 'prop-types';
import Page from '../Page';
import Form from './Form';
import { Header, Button, Divider } from 'semantic-ui-react';

const Add = ({ location, history, loading, addEmployee }) => (
  <Page title="Employee Add" location={location}>
    <Button floated="right" onClick={() => history.push('/employees')}>
      Back
    </Button>
    <Header as="h3" floated="left">
      Employee Add
    </Header>
    <Divider clearing hidden />
    <Form onSubmit={addEmployee} loading={loading} />
  </Page>
);

Add.propTypes = {
  location: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  loading: PropTypes.bool,
  addEmployee: PropTypes.func.isRequired
};

export default Add;
