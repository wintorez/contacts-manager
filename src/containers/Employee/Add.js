import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import Add from '../../components/Employee/Add';
import * as actions from '../../actions';

const mapStateToProps = (state, ownProps) => ({
  loading: state.status.loading
});

const mapDispatchToProps = (dispatch, ownProps) => ({
  addEmployee: employee =>
    dispatch(actions.addEmployee(employee)).then(({ value }) =>
      ownProps.history.push(`/employees/${value.employeeId}`)
    )
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Add)
);
