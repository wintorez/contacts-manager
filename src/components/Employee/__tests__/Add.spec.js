import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Add from '../Add';

const defaults = {
  history: {},
  addEmployee: jest.fn(),
  location: { pathname: '/employees/add' }
};

describe('<Add />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Add {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Add {...defaults} loading />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
