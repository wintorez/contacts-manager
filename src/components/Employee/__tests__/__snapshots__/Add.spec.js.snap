// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Add /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <button
      className="ui right floated button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Back
    </button>
    <h3
      className="ui left floated header"
    >
      Employee Add
    </h3>
    <div
      className="ui clearing hidden divider"
    />
    <form
      action={undefined}
      className="ui form"
      disabled={undefined}
      onSubmit={[Function]}
    >
      <div
        className="ui stackable two column grid"
      >
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                First Name
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="firstName"
                  onChange={[Function]}
                  placeholder="First Name"
                  required={true}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                Last Name
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="lastName"
                  onChange={[Function]}
                  placeholder="Last Name"
                  required={true}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                Email
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  tabIndex={undefined}
                  type="email"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="field"
            >
              <label
                htmlFor={undefined}
              >
                Telephone
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="telephone"
                  onChange={[Function]}
                  placeholder="Telephone"
                  required={undefined}
                  tabIndex={undefined}
                  type="tel"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <label
                htmlFor={undefined}
              >
                Department
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="department"
                  onChange={[Function]}
                  placeholder="Department"
                  required={undefined}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <button
              className="ui button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="submit"
            >
              Save
            </button>
             
            or
             
            <a
              href="/employees"
              onClick={[Function]}
            >
              Cancel
            </a>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Add /> renders correctly when loading 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    <button
      className="ui right floated button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
    >
      Back
    </button>
    <h3
      className="ui left floated header"
    >
      Employee Add
    </h3>
    <div
      className="ui clearing hidden divider"
    />
    <form
      action={undefined}
      className="ui form"
      disabled={true}
      onSubmit={[Function]}
    >
      <div
        className="ui stackable two column grid"
      >
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                First Name
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="firstName"
                  onChange={[Function]}
                  placeholder="First Name"
                  required={true}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                Last Name
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="lastName"
                  onChange={[Function]}
                  placeholder="Last Name"
                  required={true}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="required field"
            >
              <label
                htmlFor={undefined}
              >
                Email
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="email"
                  onChange={[Function]}
                  placeholder="Email"
                  required={true}
                  tabIndex={undefined}
                  type="email"
                />
              </div>
            </div>
          </div>
          <div
            className="column"
          >
            <div
              className="field"
            >
              <label
                htmlFor={undefined}
              >
                Telephone
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="telephone"
                  onChange={[Function]}
                  placeholder="Telephone"
                  required={undefined}
                  tabIndex={undefined}
                  type="tel"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <div
              className="field"
            >
              <label
                htmlFor={undefined}
              >
                Department
              </label>
              <div
                className="ui input"
                content={undefined}
              >
                <input
                  defaultValue=""
                  disabled={undefined}
                  name="department"
                  onChange={[Function]}
                  placeholder="Department"
                  required={undefined}
                  tabIndex={undefined}
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="column"
          >
            <button
              className="ui loading button"
              disabled={undefined}
              onClick={[Function]}
              role="button"
              tabIndex={undefined}
              type="submit"
            >
              Save
            </button>
             
            or
             
            <a
              href="/employees"
              onClick={[Function]}
            >
              Cancel
            </a>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
